Formula/opencv@3.rb

```diff
@@ -83,26 +99,24 @@ def install
       -DWITH_VTK=OFF
       -DBUILD_opencv_python2=OFF
       -DBUILD_opencv_python3=ON
-      -DPYTHON3_EXECUTABLE=#{Formula["python@3.9"].opt_bin}/python3
+      -DPYTHON3_EXECUTABLE=#{which("python3")}
     ]
 
-    if Hardware::CPU.intel?
+    if Hardware::CPU.intel? && build.bottle?
       args << "-DENABLE_AVX=OFF" << "-DENABLE_AVX2=OFF"
       args << "-DENABLE_SSE41=OFF" << "-DENABLE_SSE42=OFF" unless MacOS.version.requires_sse42?
     end
 
-    mkdir "build" do
-      system "cmake", "..", *args
-      inreplace "modules/core/version_string.inc", Superenv.shims_path, ""
-      system "make"
-      system "make", "install"
-      system "make", "clean"
-      system "cmake", "..", "-DBUILD_SHARED_LIBS=OFF", *args
-      inreplace "modules/core/version_string.inc", Superenv.shims_path, ""
-      system "make"
-      lib.install Dir["lib/*.a"]
-      lib.install Dir["3rdparty/**/*.a"]
-    end
+    system "cmake", "-S", ".", "-B", "build_shared", *args
+    inreplace "build_shared/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
+    system "cmake", "--build", "build_shared"
+    system "cmake", "--install", "build_shared"
+
+    system "cmake", "-S", ".", "-B", "build_static", *args, "-DBUILD_SHARED_LIBS=OFF"
+    inreplace "build_static/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
+    system "cmake", "--build", "build_static"
+    lib.install Dir["build_static/lib/*.a"]
+    lib.install Dir["build_static/3rdparty/**/*.a"]
```

Comment:
```suggestion
    lib.install buildpath.glob("build_static/{lib,3rdparty/**}/*.a")
```

Reply:
On side note, upstream ends up installing 3rd party libraries in odd location for v3.x https://github.com/opencv/opencv/blob/3.4.16/cmake/OpenCVInstallLayout.cmake#L67
```cmake
  ocv_update(OPENCV_3P_LIB_INSTALL_PATH        "share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH}")
```

And a more standard path in v4.x https://github.com/opencv/opencv/blob/4.6.0/cmake/OpenCVInstallLayout.cmake#L67
```cmake
  ocv_update(OPENCV_3P_LIB_INSTALL_PATH        "${OPENCV_LIB_INSTALL_PATH}/opencv4/3rdparty")
```

---

Probably can keep existing path to avoid breaking path for users. As long as there are no name conflicts with other formulae, shouldn't be an issue.